apps/aecore/src/aec_conductor.erl

```diff
@@ -229,8 +234,32 @@ handle_cast(Other, State) ->
     epoch_mining:error("Received unknown cast: ~p", [Other]),
     {noreply, State}.
 
+handle_info({gproc_ps_event, candidate_block, _}, State = #state{ mining_state = stopped }) ->
+    %% ignore new candidates if we are not running.
+    {noreply, State};
+handle_info({gproc_ps_event, candidate_block, #{info := new_candidate}}, State) ->
+    case State#state.block_candidate of
+        undefined ->
+            case aec_block_generator:get_candidate() of
+                {ok, Block} ->
+                    Candidate = make_candidate(Block, State#state.seen_top_block_hash),
+                    State1 = State#state{ block_candidate = Candidate },
+                    {noreply, start_mining(State1)};
+                {error, no_candidate} ->
+                    {noreply, State}
+            end;
+        #candidate{} ->
+            State1 = State#state{ new_candidate_available = true },
+            {noreply, State1}
+    end;
 handle_info(init_continue, State) ->
     %% Continue the initialization by (possibly) starting the miner
+    case State#state.mining_state of
+        running ->
+            aec_block_generator:start_generation();
```

Comment:
Assuming `autostart` option, process `aec_conductor` in `init_continue` calls `gen_server:cast(aec_block_generator, start_generation)` that returns `ok` even if `aec_block_generator` not registered then waits for event `candidate_block`. Considering that process `aec_block_candidate` initializes with `generating = false` and that `aecore_sup` uses `one_for_one` strategy, if process `aec_block_generator` terminated abnormally after `aec_conductor` had called `gen_server:cast(aec_block_generator, start_generation)` then process `aec_conductor` would wait forever (I do not see timeout in `aec_conductor` for `candidate_block` event though I did not thoroughly look for it).

Reply:
I would expect `aecore_sup` supervision strategy refined with intermediate `rest_for_one` with children - in the order - `aec_block_generator` and `aec_conductor`.

Reply:
Great catch, I was worried about the opposite scenario when the conductor crashed and made sure that it worked. The opposite totally slipped my mind... I have refined the supervision tree (and updated the graph accordingly!))